ballerina/modules/bulk/client.bal

```diff
@@ -39,30 +39,26 @@ public isolated client class Client {
     # + return - An error on failure of initialization or else `()`
     public isolated function init(ConnectionConfig config) returns error? {
         http:ClientConfiguration httpClientConfig = check config:constructHTTPClientConfig(config);
-        http:OAuth2RefreshTokenGrantConfig|http:BearerTokenConfig auth = let var authConfig = config.auth in 
-                (authConfig is http:BearerTokenConfig ?  authConfig : {...authConfig});
+        http:OAuth2RefreshTokenGrantConfig|http:BearerTokenConfig auth = let var authConfig = config.auth in
+                (authConfig is http:BearerTokenConfig ? authConfig : {...authConfig});
         self.clientConfig = auth.cloneReadOnly();
 
-        http:ClientOAuth2Handler|http:ClientBearerTokenAuthHandler|error httpHandlerResult;
-        if auth is http:OAuth2RefreshTokenGrantConfig {
-            httpHandlerResult = trap new http:ClientOAuth2Handler(auth);
-        } else {
-            httpHandlerResult = trap new http:ClientBearerTokenAuthHandler(auth);
-        }
+        http:ClientOAuth2Handler|http:ClientBearerTokenAuthHandler httpHandlerResult;
 
-        if httpHandlerResult is http:ClientOAuth2Handler|http:ClientBearerTokenAuthHandler {
-            self.clientHandler = httpHandlerResult;
+        if auth is http:OAuth2RefreshTokenGrantConfig {
+            httpHandlerResult = new http:ClientOAuth2Handler(auth);
         } else {
-            return error(INVALID_CLIENT_CONFIG);
+            httpHandlerResult = new http:ClientBearerTokenAuthHandler(auth);
```

Comment:
In ClientOAuth2Provider initialization, we panic[1]. So, we need to trap the error.

1. https://github.com/ballerina-platform/module-ballerina-oauth2/blob/master/ballerina/client_oauth2_provider.bal#L207